### PR TITLE
MINIFICPP-1823 Fix absolute.path output attribute in ListFile

### PR DESCRIPTION
**Pull request by lordgamez**
Adjusted absolute.path attribute according to the output description and the FetchFile expectation.

https://issues.apache.org/jira/browse/MINIFICPP-1823

-------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/standard-processors/processors/ListFile.cpp [hunk 1]
```diff
@@ -234,18 +234,19 @@ void ListFile::onTrigger(const std::shared_ptr<core::ProcessContext> &context, c
   auto file_list = utils::file::FileUtils::list_dir_all(input_directory_, logger_, recurse_subdirectories_);
   for (const auto& [path, filename] : file_list) {
     ListedFile listed_file;
-    listed_file.absolute_path = (std::filesystem::path(path) / filename).string();
+    listed_file.full_file_path = (std::filesystem::path(path) / filename).string();
+    listed_file.absolute_path = path + utils::file::FileUtils::get_separator();
```
Comment (szaszm) [inline extensions/standard-processors/processors/ListFile.cpp:238]: I don't think absolute_path has to end with a separator.

Comment (szaszm) [reply extensions/standard-processors/processors/ListFile.cpp:238]: Nevermind, just checked the nifi docs and they do it the same way.

**Comment by adamdebreceni** [comment]
note: it seems to me that both `TailFile` and `FetchS3Object` treat `"absolute.path"` as the path to the file itself, not to its directory

**Comment by lordgamez** [comment]
> note: it seems to me that both `TailFile` and `FetchS3Object` treat `"absolute.path"` as the path to the file itself, not to its directory

Yes, that was the reason of this bug that I expected that it works the same way here, but NiFi also has this difference between processors.

**Pull request closed without merging**